daceml/onnx/op_implementations/pure_implementations.py

```diff
@@ -62,6 +62,463 @@ def mulop(A: atype, B: btype, C: ctype):
+        state_exp.add_edge(terf, '_b', mx, None,
+                           dace.Memlet.simple(output, 'i, j, k'))
+        state_exp.add_edge(
+            mx, None, output, None,
+            dace.Memlet.simple(output, '0:' + M + ', 0:' + N + ', 0:' + K))
+        sdfg_exp.fill_scope_connectors()
+        return sdfg_exp
+
+@autoregister_params(op="Sqrt")
+class PureSqrt(ONNXForward):
+    @staticmethod
+    def forward_can_be_applied(node: ONNXOp, state: SDFGState,
+                               sdfg: SDFG) -> bool:
+        return True
+
+    @staticmethod
+    def forward(node: ONNXOp, state: SDFGState,
+                sdfg: SDFG) -> typing.Union[Node, SDFG]:
+
+        node.validate(sdfg, state)
+
+        in_edges = state.in_edges(node)
+        out_edges = state.out_edges(node)
+
+        atype = copy.deepcopy(sdfg.arrays[in_edges[0].data.data])
+        btype = copy.deepcopy(sdfg.arrays[out_edges[0].data.data])
+
+        @dace.program
+        def sqrtop(X: atype, Y: btype):
+            # Y[:] = X ** dace.float32(0.5)
+            Y[:] = sqrt(X)
+
+        return sqrtop.to_sdfg()
+
+@autoregister_params(op="Pow")
+class PurePow(ONNXForward):
+    @staticmethod
+    def forward_can_be_applied(node: ONNXOp, state: SDFGState,
+                               sdfg: SDFG) -> bool:
+        return True
+
+    @staticmethod
+    def forward(node: ONNXOp, state: SDFGState,
+                sdfg: SDFG) -> typing.Union[Node, SDFG]:
+
+        node.validate(sdfg, state)
+
+        in_edges = state.in_edges(node)
+        out_edges = state.out_edges(node)
+
+        atype = copy.deepcopy(sdfg.arrays[in_edges[0].data.data])
+        btype = copy.deepcopy(sdfg.arrays[in_edges[1].data.data])
+        ctype = copy.deepcopy(sdfg.arrays[out_edges[0].data.data])
+
+        @dace.program
+        def powop(X: atype, Y: btype, Z: ctype):
+            Z[:] = X ** Y
+
+        return powop.to_sdfg()
+
+@autoregister_params(op="Reshape")
+class PureReshape(ONNXForward):
+    @staticmethod
+    def forward_can_be_applied(node: ONNXOp, state: SDFGState,
+                               sdfg: SDFG) -> bool:
+        in_edges = state.in_edges(node)
+        out_edges = state.out_edges(node)
+        input_dim = len(in_edges[0].data.subset.size())
+        output_dim = len(out_edges[0].data.subset.size())
+        if input_dim == 4 and output_dim == 3:
+            return True
+        if input_dim == 3 and output_dim == 4:
+            return True
+
+        return False
+
+    @staticmethod
+    def forward(node: ONNXOp, state: SDFGState,
+                sdfg: SDFG) -> typing.Union[Node, SDFG]:
+
+        node.validate(sdfg, state)
+        
+        in_edges = state.in_edges(node)
+        out_edges = state.out_edges(node)
+
+        input_dim = len(in_edges[0].data.subset.size())
+        output_dim = len(out_edges[0].data.subset.size())
+        sdfg_exp = dace.SDFG('ReshapeExpansion')
+
+        if input_dim == 4 and output_dim == 3:
```

Comment:
Sidenote: I think it would be useful to implement this op more generally; it is used quite often.